locale/en/docs/guides/backpressuring-in-streams.md

```diff
@@ -0,0 +1,264 @@
+indefinite buffering, and the garbage collector will only have to deal with
+one area in memory!
+
+So, if backpressure is so important, why have you (probably) not heard of it? 
+Well the answer is simple: Node.js does all of this automatically for you. 
+
+That's so great! But also not so great when we are trying to understand how to 
+implement our own custom streams.
+
+## Lifecycle of `.pipe()`
+
+To achieve a better understanding of backpressure, here is a flow-chart on the 
+lifecycle of a [`Readable`][] stream being [piped][] into a [`Writable`][] 
+stream: 
+
+```javascript
++===============+
+|   Your_Data   |
++=======+=======+
+        |
++-------v-----------+          +-------------------+         +=================+
+|  Readable Stream  |          |  Writable Stream  +--------->  .write(chunk)  |
++-------+-----------+          +---------^---------+         +=======+=========+
+        |                                |                           |
+        |     +======================+   |        +------------------v---------+
+        +----->  .pipe(destination)  >---+        |    Is this chunk too big?  |
+              +==^=======^========^==+            |    Is the queue busy?      |
+                 ^       ^        ^               +----------+-------------+---+
+                 |       |        |                          |             |
+                 |       |        |  > if (!chunk)           |             |
+                 ^       |        |      emit .end();        |             |
+                 ^       ^        |  > else                  |             |
+                 |       ^        |      emit .write();  +---v---+     +---v---+
+                 |       |        ^----^-----------------<  No   |     |  Yes  |
+                 ^       |                               +-------+     +---v---+
+                 ^       |                                                 |
+                 |       ^   emit .pause();        +=================+     |
+                 |       ^---^---------------------+  return false;  <-----+---+
+                 |                                 +=================+         |
+                 |                                                             |
+                 ^   when queue is empty   +============+                      |
+                 ^---^-----------------^---<  Buffering |                      |
+                     |                     |============|                      |
+                     +> emit .drain();     |  <Buffer>  |                      |
+                     +> emit .resume();    +------------+                      |
+                                           |  <Buffer>  |                      |
+                                           +------------+  add chunk to queue  |
+                                           |            <--^-------------------<
+                                           +============+
+```
```

Comment:
@jessicaquynh it is incorrect. `pipe` sets up some closures, who would keep working, but the method is not invoked anymore. The diagram should represent what `pipe` set up, but not show `pipe` at all: https://github.com/nodejs/node/blob/master/lib/_stream_readable.js#L473-L613